app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt

```diff
@@ -6,35 +6,39 @@ import org.p2p.solanaj.utils.crypto.decodeFromBase64
 import org.p2p.wallet.auth.gateway.api.request.OtpMethod
 import org.p2p.wallet.auth.gateway.api.response.ConfirmRestoreWalletResponse
 import org.p2p.wallet.auth.gateway.repository.GatewayServiceRepository
+import org.p2p.wallet.auth.model.PhoneNumber
+import org.p2p.wallet.auth.model.RestoreWalletFailure
 import org.p2p.wallet.auth.repository.RestoreFlowDataLocalRepository
+import org.p2p.wallet.auth.repository.UserSignUpDetailsStorage
 import org.p2p.wallet.auth.web3authsdk.response.Web3AuthSignUpResponse
 import org.p2p.wallet.utils.fromJsonReified
 import timber.log.Timber
 
 class CustomShareRestoreInteractor(
     private val gatewayServiceRepository: GatewayServiceRepository,
     private val restoreFlowDataLocalRepository: RestoreFlowDataLocalRepository,
+    private val signUpDetailsStorage: UserSignUpDetailsStorage,
     private val gson: Gson,
 ) {
-    class RestoreWalletFailure(override val message: String) : Throwable(message)
 
-    suspend fun startRestoreCustomShare(userPhoneNumber: String) {
+    suspend fun startRestoreCustomShare(userPhoneNumber: PhoneNumber, isResend: Boolean = false) {
         val temporaryUserPublicKey = restoreFlowDataLocalRepository.userRestorePublicKey
             ?: throw RestoreWalletFailure("User restore public key is null")
         val temporaryUserPrivateKey = restoreFlowDataLocalRepository.userRestorePrivateKey
             ?: throw RestoreWalletFailure("User restore private key is null")
 
-        gatewayServiceRepository.restoreWallet(
-            solanaPublicKey = temporaryUserPublicKey,
-            solanaPrivateKey = temporaryUserPrivateKey,
-            phoneNumber = userPhoneNumber,
-            channel = OtpMethod.SMS
-        )
-
+        if (isResend || userPhoneNumber != restoreFlowDataLocalRepository.userPhoneNumber) {
+            gatewayServiceRepository.restoreWallet(
+                solanaPublicKey = temporaryUserPublicKey,
+                solanaPrivateKey = temporaryUserPrivateKey,
+                phoneNumber = userPhoneNumber,
+                channel = OtpMethod.SMS
+            )
+        }
         restoreFlowDataLocalRepository.userPhoneNumber = userPhoneNumber
     }
 
-    suspend fun finishRestoreCustomShare(smsCode: String) {
```

Comment:
revert this please, it's CustomShareRestoreInteractor, so it's methods should be called accordingly 
on the domain layer - we restore custom share
on data layer - we use restore / confirm_restore_wallet methods for Gateway